Sources/Signal.swift

```diff
@@ -285,8 +295,61 @@ public final class Signal<Value, Error: Swift.Error> {
 		}
 	}
 
+	private init(_ core: CoreBase) {
+		self.core = core
+	}
+
+	/// Initialize a Signal that will immediately invoke the given generator,
+	/// then forward events sent to the given input observer.
+	///
+	/// The input observer serializes events it received. In other words, it is thread safe, and
+	/// can be called on multiple threads concurrently.
+	///
+	/// - note: The disposable returned from the closure will be automatically
+	///         disposed if a terminating event is sent to the observer. The
+	///         Signal itself will remain alive until the observer is released.
+	///
+	/// - parameters:
+	///   - generator: A closure that accepts an implicitly created observer
+	///                that will act as an event emitter for the signal.
+	public convenience init(_ generator: (Observer, Lifetime) -> Void) {
+		self.init(Core<Lock>(generator))
+	}
+
+	/// Initialize a Signal that will immediately invoke the given generator,
+	/// then forward events sent to the given non-serializing input observer.
+	///
+	/// Unlike `init(_:)`, the provided input observer does not serialize the events it received, with the assumption
+	/// that it can inherit mutual exclusion from its callers.
+	///
+	/// Note that the created `Signal` still guarantees thread safe observer manipulation.
+	///
+	/// - warning: The input observer **is not thread safe**.
+	///
+	/// - note: The disposable returned from the closure will be automatically
+	///         disposed if a terminating event is sent to the observer. The
+	///         Signal itself will remain alive until the observer is released.
+	///
+	/// - parameters:
+	///   - generator: A closure that accepts an implicitly created observer
+	///                that will act as an event emitter for the signal.
+	public static func nonSerializing(_ generator: (Observer, Lifetime) -> Void) -> Signal {
```

Comment:
"Non-serialising" reads a bit awkwardly to me. How about "unserialized"? It also shares a prefix with "unsafe", which feels at least helpfully coincidental.